tardisbase/testing/regression_comparison/git_utils.py

```diff
@@ -0,0 +1,202 @@
+    """
+    if repo is None and repo_path is not None:
+        repo = Repo(repo_path)
+    elif repo is None:
+        raise ValueError("Either repo or repo_path must be provided")
+
+    try:
+        commit = repo.commit(commit_hash)
+        return {
+            'hash': commit_hash[:8],
+            'message': commit.message.strip().split('\n')[0][:60],  # First line, max 60 chars
+            'author': commit.author.name,
+            'date': commit.committed_datetime.strftime('%Y-%m-%d %H:%M')
+        }
+    except Exception as e:
+        error_msg = ERROR_MESSAGES["commit_info_failed"].format(commit=commit_hash[:8], error=str(e))
+        logger.warning(error_msg)
+        return {
+            'hash': commit_hash[:8],
+            'message': 'Unable to fetch commit info',
+            'author': 'Unknown',
+            'date': 'Unknown'
+        }
+
+
+def is_file_modified(file_path, prev_commit, current_commit, repo_path):
+    """
+    Check if file was modified between commits using git diff with improved error handling.
+    
+    Parameters
+    ----------
+    file_path : str
+        Path to the file to check
+    prev_commit : str
+        Previous commit hash
+    current_commit : str
+        Current commit hash
+    repo_path : str or Path
+        Path to the repository
+        
+    Returns
+    -------
+    bool
+        True if file was modified, False otherwise
+    """
+    try:
+        result = subprocess.run([
+            'git', 'diff', '--quiet',
+            f'{prev_commit}..{current_commit}',
+            '--', file_path
+        ], cwd=repo_path, capture_output=True)
+        return result.returncode != 0  # 0 = no diff, 1 = has diff
+    except Exception as e:
+        error_msg = ERROR_MESSAGES["git_diff_failed"].format(
+            file=file_path, 
+            commit1=prev_commit[:8], 
+            commit2=current_commit[:8],
+            error=str(e)
+        )
+        logger.warning(error_msg)
+        return True  # Assume modified if git diff fails
+
+
+def get_file_change_details(file_path, prev_commit, current_commit, repo_path):
+    """
+    Get detailed information about what changed in a file with improved error handling.
+    
+    Parameters
+    ----------
+    file_path : str
+        Path to the file
+    prev_commit : str
+        Previous commit hash
+    current_commit : str
+        Current commit hash
+    repo_path : str or Path
+        Path to the repository
+        
+    Returns
+    -------
+    str
+        Description of the changes
+    """
+    try:
+        # Get git diff stats
+        result = subprocess.run([
+            'git', 'diff', '--stat',
+            f'{prev_commit}..{current_commit}',
+            '--', file_path
+        ], cwd=repo_path, capture_output=True, text=True)
+
+        if result.returncode == 0 and result.stdout.strip():
+            return result.stdout.strip()
+        else:
+            return "File changed (no diff details available)"
```

Comment:
here too perhaps?
if we are using the module, these functionalities should be there